ocaml/typing/typecore.ml

```diff
@@ -504,9 +501,17 @@ let register_allocation_value_mode mode =
   register_allocation_mode alloc_mode;
   alloc_mode
 
+let register_allocation_value_mode' mode =
+  let alloc_mode = register_allocation_value_mode mode in
+  alloc_mode, Value_mode.of_alloc alloc_mode
+
 let register_allocation (expected_mode : expected_mode) =
   register_allocation_value_mode expected_mode.mode
 
+let register_allocation' (expected_mode : expected_mode) =
```

Comment:
Another terrible name. Why not just call this `register_allocation` and replace uses of the old `register_allocation` with calls to `register_allocation_value_mode`?